File: payload/reggae/reggae_json_build.lua
#!/usr/bin/env lua

local reggae = require('reggae')

function get_build()
   local reggaefile = require('reggaefile')
   local builds = {}
   for k, v in pairs(reggaefile) do
      if v.isBuild then
         table.insert(builds, v)
      end
   end

   if builds[2] then
      error 'Only one Build object allowed per file'
   end

   if not builds[1] then
      error 'Could not find a Build object'
   end

   return builds[1]
end

print(get_build():to_json())

File: payload/reggae/reggae_json_build.js
#!/usr/bin/env node

var reggae = require('reggae-js')

function getBuild() {
    var reggaefile = require('reggaefile')
    var builds = []

    for(var key in reggaefile) {
        if(reggaefile[key].constructor == reggae.Build) builds.push(reggaefile[key])
    }

    if(builds.length > 1) throw "Too many Build objects"
    if(builds.length == 0)
        throw "Could not find Build object in:\n" + reggaefile.toSource()

    return builds[0]
}


console.log(getBuild().toJson())
